modules/mono/editor/GodotTools/GodotTools/Inspector/InspectorPlugin.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using Godot;
+using GodotTools.Utils;
+
+namespace GodotTools.Inspector
+{
+    public partial class InspectorPlugin : EditorInspectorPlugin
+    {
+        public override bool _CanHandle(GodotObject godotObject)
+        {
+            foreach (var script in EnumerateScripts(godotObject))
+            {
+                if (script is CSharpScript)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public override void _ParseBegin(GodotObject godotObject)
+        {
+            foreach (var script in EnumerateScripts(godotObject))
+            {
+                if (script is not CSharpScript) continue;
+
+                if (File.GetLastWriteTime(script.ResourcePath) > GodotSharpEditor.Instance.LastValidBuildDateTime)
+                {
+                    AddCustomControl(new InspectorOutOfSyncWarning());
+                    break;
+                }
+            }
+        }
+
+        private static IEnumerable<Script> EnumerateScripts(GodotObject godotObject)
+        {
+            var script = godotObject.GetScript().AsGodotObject() as Script;
```

Comment:
I imagine the result of `GetScript` will always be a `Script`. I think it only returns `Variant` because it can be null, but in that case C# won't throw an exception.

```suggestion
            var script = godotObject.GetScript().As<Script>();
```